onnxruntime/core/optimizer/qdq_transformer/selectors_actions/shared/utils.cc

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <iostream>
+#include <string>
+#include <vector>
+#include <optional>
+
+#include <core/graph/graph_viewer.h>
+#include <core/providers/common.h>
+
+#include "core/optimizer/qdq_transformer/selectors_actions/qdq_selector_action_transformer.h"
+#include "core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.h"
+
+#include "utils.h"
+
+namespace onnxruntime {
+namespace QDQ {
+
+using std::string;
+using std::vector;
+
+using OpVersionsMap = std::unordered_map<std::string, std::vector<ONNX_NAMESPACE::OperatorSetVersion>>;
+
+void Selectors::RegisterSelector(const Selector::OpVersionsMap& ops_and_versions_in,
+                                 std::unique_ptr<QDQ::BaseSelector> selector_in) {
+  auto entry = std::make_unique<Selector>(
+      ops_and_versions_in,
+      std::move(selector_in));
+
+  ORT_IGNORE_RETURN_VALUE(selectors_set_.insert(std::move(entry)));
+}
+
+/* Selector Rules Related */
+void RegisterMiscSelectors(Selectors& qdq_selectors) {
+  /* register selectors for miscellaneous ops */
+  std::unique_ptr<BaseSelector> selector(new QDQ::DropDQDNodesSelector());
+  qdq_selectors.RegisterSelector(Selector::OpVersionsMap{{"Gather", {}},
+                                                         {"Reshape", {}},
+                                                         {"Transpose", {}},
+                                                         {"MaxPool", {12}},
+                                                         {"Resize", {}}},
+                                 std::move(selector));
+}
+
+void RegisterUnarySelectors(Selectors& qdq_selectors) {
+  /* regsiter selectors for unary ops */
+  std::unique_ptr<BaseSelector> selector(new QDQ::UnarySelector());
+  qdq_selectors.RegisterSelector(Selector::OpVersionsMap{{"AveragePool", {}}},
```

Comment:
This line is already updated in here
https://github.com/microsoft/onnxruntime/blob/2078210a1cc2f9defc01452300ac582b0053f8cd/onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selector_action_transformer.cc#L56-L58

To avoid the fork of the code, can we keep them in a single place for these keys?

Reply:
Updated. Add some static methods for this.

Reply:
will qdq_selector_action_transformer.cc be updated to use the same helpers?

Reply:
Things are going to diverge so I don't think we should try and keep it in sync with the CPU EP's QDQ SAT.

The QDQ node group selectors for use by other EPs are looking for all QDQ node groups. The CPU EP is looking for things it has QLinear implementations for. 

Additionally the CPU EP will be registering NodeSelector based classes in SelectorAndAction. The QDQ node group selectors will be registering NodeGroupSelector based classes. 

So they will look somewhat similar initially but I expect as we keep expanding the ops we match as QDQ node groups that won't be for long.

e.g. SoftMax should be matched in NNAPI but not the CPU EP. Similarly we currently have only Add and Mul, however (afaik) there's no reason Sub and Div shouldn't be matched and those can easily be added to the NodeGroupSelector registrations.